module/VuFind/src/VuFind/Db/Row/Search.php

```diff
@@ -160,4 +171,236 @@ public function getUnsubscribeSecret(HMAC $hmac, $user)
         ];
         return $hmac->generate(array_keys($data), $data);
     }
+
+    /**
+     * Get identifier (returns null for an uninitialized or non-persisted object).
+     *
+     * @return ?int
+     */
+    public function getId(): ?int
+    {
+        return $this->id ?? null;
+    }
+
+    /**
+     * Get user.
+     *
+     * @return int
+     */
+    public function getUserId(): int
+    {
+        return $this->user_id
+        ? $this->getDbServiceManager()->get(UserServiceInterface::class)->getUserById($this->user_id)
+        : null;
+    }
+
+    /**
+     * Set user.
+     *
+     * @param int $user User
+     *
+     * @return SearchEntityInterface
+     */
+    public function setUserId(int $user): SearchEntityInterface
+    {
+        $this->user_id = $user->getId();
+        return $this;
+    }
+
+    /**
+     * Get session.
+     *
+     * @return ?string
+     */
+    public function getSessionId(): ?string
+    {
+        return $this->session_id ?? null;
+    }
+
+    /**
+     * Set session.
+     *
+     * @param ?string $session_id Session id
+     *
+     * @return SearchEntityInterface
+     */
+    public function setSessionId(?string $session_id): SearchEntityInterface
+    {
+        $this->session_id = $session_id;
+        return $this;
+    }
+
+    /**
+     * Get created.
+     *
+     * @return DateTime
+     */
+    public function getCreated(): DateTime
+    {
+        return DateTime::createFromFormat('Y-m-d H:i:s', $this->created);
+    }
+
+    /**
+     * Set created date.
+     *
+     * @param DateTime $dateTime Created date
+     *
+     * @return SearchEntityInterface
+     */
+    public function setCreated(DateTime $dateTime): SearchEntityInterface
+    {
+        $this->created = $dateTime->format('Y-m-d H:i:s');
+        return $this;
+    }
+
+    /**
+     * Get title.
+     *
+     * @return ?string
+     */
+    public function getTitle(): ?string
+    {
+        return $this->title ?? null;
+    }
+
+    /**
+     * Set title.
+     *
+     * @param ?string $title Title
+     *
+     * @return SearchEntityInterface
+     */
+    public function setTitle(?string $title): SearchEntityInterface
+    {
+        $this->title = $title;
+        return $this;
+    }
+
+    /**
+     * Get saved.
+     *
+     * @return int
+     */
+    public function getSaved(): int
+    {
+        return $this->saved ?? 0;
+    }
+
+    /**
+     * Set saved.
+     *
+     * @param int $saved Saved
+     *
+     * @return SearchEntityInterface
+     */
+    public function setSaved(int $saved): SearchEntityInterface
+    {
+        $this->saved = $saved;
+        return $this;
+    }
+
+    /**
+     * Set search object.
+     *
+     * @param \VuFind\Search\Minified $search_object Search object
+     *
+     * @return SearchEntityInterface
+     */
+    public function setSearchObject(\VuFind\Search\Minified $search_object): SearchEntityInterface
+    {
+        $this->search_object = $search_object;
```

Comment:
Since we are unserializing this in the getter, we need to `serialize()` it here in the setter to keep things symmetrical.